src/Common/src/TypeSystem/Interop/IL/MarshalHelpers.cs

```diff
@@ -9,56 +9,8 @@
 
 namespace Internal.TypeSystem.Interop
 {
-    public static class MarshalHelpers
+    public static partial class MarshalHelpers
```

Comment:
I see you investigated just making the class partial. Why not go with that and just put the `IsBlittableType` in a dot file?

`IsBlittable.CheckType` is less consistent with the naming scheme we use elsewhere than `MarshalHelpers.IsBlittableType`. (Name of the method doesn't capture what the method does.)

Reply:
I like your suggestion, I'll modify the change as you proposed.

Reply:
Hmm, this turns out to be quite a viral change; the problem is that, once an assembly, in particular System.Private.TypeLoader, includes the partial class from the MarshalHelpers.IsBlittableType.cs file, it must include complete source code for the class, otherwise any other assembly including MarshalHelpers.cs and depending on S.P.TypeLoader complains that the type is ambiguous.

Reply:
In 2nd commit I have opted for a compromise - I haven't made MarshalHelpers partial as it would require adding the complete marshalling source code to System.Private.TypeLoader; instead I renamed the check to MarshalUtils.IsBlittableType. Please let me know if you see a cleaner way to approach this.